File: FrontEnd/src/services/reviewService.js
import api from '../api/axiosInstance';

const ReviewService = {
  getProductReviews: (productId, params = {}) =>
    api.get(`/products/${productId}/reviews`, { params }),

  getProductReviewSummary: (productId) =>
    api.get(`/products/${productId}/reviews/summary`),

  canUserReview: (productId, userId) =>
    api.get(`/products/${productId}/reviews/can-review`, { params: { userId } }),

  createReview: (userId, reviewData) =>
    api.post('/reviews', reviewData, { params: { userId } }),

  updateReview: (reviewId, userId, reviewData) =>
    api.put(`/reviews/${reviewId}`, reviewData, { params: { userId } }),

  deleteReview: (reviewId, userId) =>
    api.delete(`/reviews/${reviewId}`, { params: { userId } }),

  markHelpful: (reviewId) =>
    api.post(`/reviews/${reviewId}/helpful`),

  getUserReviews: (userId) =>
    api.get(`/reviews/user/${userId}`),
};

export default ReviewService;

File: BackEnd/phone-store-backend/src/main/java/com/tmdt/phone_store_backend/service/ReviewService.java
package com.tmdt.phone_store_backend.service;

import com.tmdt.phone_store_backend.dto.AdminReviewDto;
import com.tmdt.phone_store_backend.dto.CreateReviewRequestDto;
import com.tmdt.phone_store_backend.dto.PagedAdminReviewResponseDto;
import com.tmdt.phone_store_backend.dto.PagedReviewResponseDto;
import com.tmdt.phone_store_backend.dto.ProductReviewSummaryDto;
import com.tmdt.phone_store_backend.dto.ReviewDto;
import com.tmdt.phone_store_backend.dto.UpdateReviewRequestDto;
import java.util.List;

public interface ReviewService {

    PagedReviewResponseDto getProductReviews(Long productId, int page, int size, String sortBy, Integer filterRating);

    ProductReviewSummaryDto getProductReviewSummary(Long productId);

    ReviewDto createReview(Long userId, CreateReviewRequestDto requestDto);

    ReviewDto updateReview(Long userId, Long reviewId, UpdateReviewRequestDto requestDto);

    void deleteReview(Long userId, Long reviewId);

    void markReviewHelpful(Long reviewId);

    boolean canUserReviewProduct(Long userId, Long productId);

    boolean hasUserPurchasedProduct(Long userId, Long productId);

    boolean hasUserPendingReview(Long userId, Long productId);

    boolean existsReviewByUserAndProduct(Long userId, Long productId);

    List<ReviewDto> getUserReviews(Long userId);

    ReviewDto getReviewById(Long reviewId);

    // Admin methods
    PagedAdminReviewResponseDto getPendingReviews(int page, int size);

    PagedAdminReviewResponseDto getAllReviews(int page, int size, Long productId, Boolean approved);

    AdminReviewDto approveReview(Long reviewId);

    AdminReviewDto rejectReview(Long reviewId);

    void deleteReviewAdmin(Long reviewId);
}
